flopy/modflow/mfbas.py

```diff
@@ -215,16 +215,29 @@ def check(self, f=None, verbose=True, level=1, checktype=None):
         """
         chk = self._get_check(f, verbose, level, checktype)
 
-        neighbors = get_neighbors(self.ibound.array)
-        neighbors[
-            np.isnan(neighbors)
-        ] = 0  # set neighbors at edges to 0 (inactive)
-        chk.values(
-            self.ibound.array,
-            (self.ibound.array > 0) & np.all(neighbors < 1, axis=0),
-            "isolated cells in ibound array",
-            "Warning",
-        )
+        if self.parent.structured:
+            neighbors = get_neighbors(self.ibound.array)
+            neighbors[
+                np.isnan(neighbors)
+            ] = 0  # set neighbors at edges to 0 (inactive)
+            chk.values(
+                self.ibound.array,
+                (self.ibound.array > 0) & np.all(neighbors < 1, axis=0),
+                "isolated cells in ibound array",
+                "Warning",
+            )
+        elif "DISU" in self.parent.get_package_list():
+            neighbors = get_neighbors_u(self.ibound.array, self.parent.disu)
+            n_neighbors_active = np.array(
+                [len(n[n != 0]) for n in neighbors]
+            )  # including CHDs == ok.
+            chk.values(
+                self.ibound.array,
+                (self.ibound.array != 0) & (n_neighbors_active == 0),
+                "isolated cells in ibound array",
+                "Warning",
+            )
```

Comment:
This is where things are going to start getting messy.  Are we going to need to add structured and unstructured code to all these modflow packages?   Ideally, I don't think we don't want to do that as it will become unsustainable to maintain.  Perhaps a better way is to add a "get_neighbors" method to the model grid classes and make sure they return the information in the same format?